### PR TITLE
Refactoring of image processing before save

### PR DESCRIPTION
**Pull request by Riddlerrr**
task:  https://www.pivotaltracker.com/story/show/178559683

`has_one_attached_and_prepare_image` introduced
It support resize changes, like this:
`has_one_attached_and_prepare_image :image, resize: '190x190!'`
Resize options: https://legacy.imagemagick.org/Usage/resize/#noaspect

After set attachment to image we check it using `ImagePreparer` and reset attachment from it.

ImagePreparer changes:
- It now process only one image
- it save image in `.error` and set `attachment` to nil in case of errors

### CODE REVIEW

**Review by Alexey1100 — CHANGES_REQUESTED** [review]
Oops I've reviewed the https://github.com/CoMakery/comakery-server/pull/2192 first.

Pls see comments there, which mostly relate to code in this PR, or maybe we can merge them in one.

**Review by aquabu — APPROVED** [review]

**Review by aquabu — CHANGES_REQUESTED** [review]
see comment

File: app/controllers/award_types_controller.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def edit
   def create
     @award_type = @project.award_types.new(award_type_params)
 
-    if ImagePreparer.new(@award_type, award_type_params).valid? && @award_type.save
+    if @award_type.save_if_no_errors
```
[the remark this answers is not in the document]
Comment (Riddlerrr) [reply app/controllers/award_types_controller.rb:25]: Explained here https://github.com/CoMakery/comakery-server/pull/2192#discussion_r655891324

**Comment by Riddlerrr** [comment]
All these changes in https://github.com/CoMakery/comakery-server/pull/2192
Closing this PR to not confuse

**Pull request closed without merging**